### ISSUE TITLE
[Bug]: Zitadel deployment on EKS with AWS PostgreSQL RDS instance - setup job failure

### ISSUE THREAD

**Issue by jmutai** [zitadel/zitadel#8789]
### Preflight Checklist

- [x] I could not find a solution in the documentation, the existing issues or discussions
- [x] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

Self-hosted

### Version

8.5.0

### Database

PostgreSQL

### Database Version

16

### Describe the problem caused by this bug

Attempting installation of Zitadel on EKS Kubernetes cluster fails at setup job when using AWS RDS for PostgreSQL. Initially the issue was with SSL, but after setting SSL mode to disable, and `rds.force_ssl = 0` on RDS, I was able to connect successfully to RDS using username and password.

Database initialization was also successful, but setup job will get stuck to infinity with the following error:

`time="2024-10-16T18:28:34Z" level=warning msg="migration already started, will check again in 5 seconds" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:130" migration step=projections.login_policies5
`

Can you help point out if this is known issue and there is a workaround.

### To reproduce

1. Provision PostgreSQL DB instance on RDS
2. Edit Parameter Group and set rds.force_ssl = 0
3. Create a secret with RDS DB credentials - host and password
4. Deploy Zitadel on Kubernetes using Helm, while using correct values for configSecretName and configSecretKey
5. Zitadel init job will be successful
6. Setup job will not be successful

### Screenshots

![Image](https://github.com/user-attachments/assets/a1decc96-fb02-44d7-8f10-08b1d5c4ac7f)

### Expected behavior

Init and Setup job successful for Zitadel to be healthy and running.

### Operating System

Kubernetes (AWS EKS) - 1.30

### Relevant Configuration

  Database:
              Postgres:
                Port: 5432
                Database:  <rds-db-endpoint>
                MaxOpenConns: 20
                MaxIdleConns: 10
                MaxConnLifetime: 30m
                MaxConnIdleTime: 5m
                User:
                  Username: zitadel
                  SSL:
                    Mode: disable
                Admin:
                  Username: zitadel
                  SSL:
                    Mode: disable
configSecretName: zitadel-config-secret
configSecretKey: config.yaml

**Comment by suchitsancheti** [comment]
Hi,

I am also facing the same issue while upgrading the zitadel from v2.55.8 to 2.56.0. The deployment is based on EKS with Postgres

Please suggest

**Comment by suchitsancheti** [comment]
Any update/workaround on this issue?

**Comment by Smana** [comment]
The ssl mode should be `required` (I switched from RDS to CNPG but that was working with RDS).
Here is my repository: https://github.com/Smana/cloud-native-ref

**Comment by adlerhurst** [comment]
Hi the error of the screenshot can be fixed by executing `zitadel setup cleanup` cli command

**Comment by jmutai** [comment]
> The error of the screenshot can be fixed by executing `zitadel setup cleanup` cli command

We are running the installation of Zitadel in Gitops version. Is there solution to that suits to automated setup process?

**Comment by jmutai** [comment]
> The ssl mode should be `required` (I switched from RDS to CNPG but that was working with RDS). Here is my repository: https://github.com/Smana/cloud-native-ref
> 
> More specificaly [here](https://github.com/Smana/cloud-native-ref/tree/main/security/base/zitadel)

Do you have working configurations for RDS that you can share?. I will really appreciate.

**Comment by jmutai** [comment]
@Smana Do you have working configurations for RDS that you can share?. I will really appreciate.

**Comment by Smana** [comment]
@jmutai As I mentioned the trick is to keep the default sslMode for RDS but ensure that Zitadel uses `require`

These are the envvars that I source
```console
ZITADEL_DATABASE_POSTGRES_USER_SSL_MODE=require
ZITADEL_DATABASE_POSTGRES_ADMIN_SSL_MODE=require
```

And [here](https://github.com/Smana/cloud-native-ref/tree/f796beda11fa41649ad7df6b35528597d9581d5d/security/base/zitadel) is the working code (when I was using RDS).

**Comment by jmutai** [comment]
@Smana will test this out and see on how it goes. Thank you!

**Comment by dfry** [comment]
@Smana thanks for the example.  I am confused though.  You are setting the master password in the crossplane config also referencing the same envvar secret.  Does that mean that the master password is just being used for other rds postgresql admin funcionality for the database and not by zitadel?  I am missing where rds is being configured to trust client certs issued by the vault cert manager cluster issuer.

**Comment by Smana** [comment]
> I was confusing myself, now I see you are using username/password auth for connecting to RDS and not client certs.

Yes and all the configurations are loaded from environment variables.

**Comment by suchitsancheti** [comment]
After using the `zitadel setup cleanup` and env variables as mentioned above, the setup job is stuck at following:

`time="2024-11-15T14:38:08Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:43" name=23_correct_global_unique_constraints
time="2024-11-15T14:38:08Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:43" name=24_add_actor_col_to_auth_tokens
time="2024-11-15T14:38:08Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:43" name=26_auth_users3
time="2024-11-15T14:38:08Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:43" name=29_init_fields_for_project_grant
time="2024-11-15T14:38:08Z" level=info msg="starting migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:66" name=29_init_fields_for_project_grant`

Please suggest

**Comment by jmutai** [comment]
@suchitsancheti did you try increasing activeDeadlineSeconds value of setupJob?

`setupJob:
          activeDeadlineSeconds: 600`

**Comment by suchitsancheti** [comment]
@jmutai I have tried with values 600, 900 and 1200 and got the same result.
Anything else that I can try/check?
